testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestAvroSchemaStrictness.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hive;
+
+import com.google.inject.name.Named;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.hadoop.hdfs.HdfsClient;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tempto.query.QueryExecutor.param;
+import static io.trino.tempto.query.QueryExecutor.query;
+import static io.trino.tests.product.TestGroups.AVRO;
+import static io.trino.tests.product.TestGroups.HIVE_WITH_EXTERNAL_WRITES;
+import static io.trino.tests.product.hive.util.TemporaryHiveTable.randomTableSuffix;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static java.nio.file.Files.newInputStream;
+import static java.sql.JDBCType.VARCHAR;
+
+/**
+ * Avro 1.8.2 was not strict about schema validation, causing the more-strict
+ * Avro 1.9.2 to cause failures with some schemas that previously worked. This
+ * was resolved in Avro 1.10.2's reader.
+ *
+ * <p>This class tests that Trino can read some data written with invalid
+ * schemas by Avro 1.8.2.
+ */
+public class TestAvroSchemaStrictness
+        extends ProductTest
+{
+    private static final Path ILLEGAL_UNION_DEFAULT_SCHEMA = Path.of("/docker/presto-product-tests/avro/invalid_default.avsc");
+    /**
+     *  The data in the avro data file was generated from the following JSON data using avro-tools.
+     *  <pre>
+     *  {"valid": {"string": "valid"}, "invalid": {"string": "invalid"}}
+     *  {"valid": null, "invalid": null}
+     *  </pre>
+     *
+     *  The command used:
+     *  <pre>{@code
+     *  java.jar avro-tools-1.8.2.jar fromjson --schema-file "$SCHEMA_FILE" data.json > invalid_default.avsc
+     *  }</pre>
+     */
+    private static final Path ILLEGAL_UNION_DEFAULT_DATA = Path.of("/docker/presto-product-tests/avro/invalid_default.avro");
+
+    @Inject
+    private HdfsClient hdfsClient;
+
+    @Inject
+    @Named("databases.hive.warehouse_directory_path")
+    private String warehouseDirectory;
+
+    @Test(groups = {AVRO, HIVE_WITH_EXTERNAL_WRITES})
```

Comment:
I don't think HIVE_WITH_EXTERNAL_WRITES needs be here.

```suggestion
    @Test(groups = AVRO)
```

Reply:
BTW @hashhar do we actually need HIVE_WITH_EXTERNAL_WRITES group to exist?

Reply:
Hmmm, good point. Only a catalog with external writes is needed. The group seems redundant. There is one other which uses the group. We can remove it from there too (it'll run on suite-1 I guess?).